src/Pimple/Container.php

```diff
@@ -233,8 +233,8 @@ public function extend($id, $callable)
             throw new \InvalidArgumentException(sprintf('Identifier "%s" does not contain an object definition.', $id));
         }
 
-        if (!is_object($callable) || !method_exists($callable, '__invoke')) {
-            throw new \InvalidArgumentException('Extension service definition is not a Closure or invokable object.');
+        if (!is_callable($callable)) {
```

Comment:
Valid point, even though 'not possible' is more of a 'could cause too many unintended side effects'.  I can see the rationale in forbidding plain callables, but perhaps that should be documented a bit more explicitly.